.github/workflows/merge-release-pr.yml

```
@@ -0,0 +1,42 @@
+name: Merge Release PR and Publish
+
+on:
+  pull_request:
+    types:
+      - closed
```

```suggestion
  pull_request:
    types:
      - closed
      branches:
       - 'release-v*'
```

otherwise it will run on every pull request which will get closed and also check for merge status not only close otherwise it will run even if pr is marked closed instead of merge